test/test_pipeline.py

```diff
@@ -182,203 +204,320 @@ def test_to_code():
     assert index_pipeline.get_config() == locals()["index_pipeline_from_code"].get_config()
 
 
-@pytest.mark.elasticsearch
-def test_PipelineCodeGen_simple_sparse_pipeline():
-    doc_store = ElasticsearchDocumentStore(index="my-index")
-    retriever = ElasticsearchRetriever(document_store=doc_store, top_k=20)
+def test_get_config_creates_dependent_component():
+    child = ChildComponent()
+    parent = ParentComponent(dependent=child)
     pipeline = Pipeline()
-    pipeline.add_node(component=retriever, name="retri", inputs=["Query"])
+    pipeline.add_node(component=parent, name="parent", inputs=["Query"])
 
-    code = _PipelineCodeGen.generate_code(pipeline=pipeline, pipeline_variable_name="p", generate_imports=False)
-    assert code == (
-        'elasticsearch_document_store = ElasticsearchDocumentStore(index="my-index")\n'
-        "retri = ElasticsearchRetriever(document_store=elasticsearch_document_store, top_k=20)\n"
-        "\n"
-        "p = Pipeline()\n"
-        'p.add_node(component=retri, name="retri", inputs=["Query"])'
-    )
+    expected_pipelines = [{"name": "query", "type": "Pipeline", "nodes": [{"name": "parent", "inputs": ["Query"]}]}]
```

Comment:
You probably don't need `type: Pipeline`, and I plan to make it fail validation soon (see #2226 if you are curious). But double check that's actually the case.

Same for all other instances below.

Reply:
I leave it up for the "getting rid of it"-PR